pkg/compose/build.go

```diff
@@ -352,3 +358,26 @@ func addSecretsConfig(project *types.Project, service types.ServiceConfig) (sess
 	}
 	return secretsprovider.NewSecretProvider(store), nil
 }
+
+func addPlatforms(project *types.Project, service types.ServiceConfig) ([]specs.Platform, error) {
+	var plats []specs.Platform
+	if platform, ok := project.Environment["DOCKER_DEFAULT_PLATFORM"]; ok {
+		p, err := platforms.Parse(platform)
+		if err != nil {
+			return nil, err
+		}
+		plats = append(plats, p)
+	}
+	if service.Platform != "" && !utils.StringContains(service.Build.Platforms, service.Platform) {
+		return nil, fmt.Errorf("service.platform should be part of the service.build.platforms: %q", service.Platform)
+	}
+
+	for _, buildPlatform := range service.Build.Platforms {
+		p, err := platforms.Parse(buildPlatform)
+		if err != nil {
+			return nil, err
+		}
+		plats = append(plats, p)
```

Comment:
I think we need to make sure we don't add a duplicate if a platform is already present via `DOCKER_DEFAULT_PLATFORM` above:

```yaml
services:
  hello:
    build:
      context: .
      platforms:
        - linux/amd64
```

```shell
$ DOCKER_DEFAULT_PLATFORM="linux/amd64" docker compose build hello
...
 => ERROR exporting to image                                                                                                                           0.0s
------
 > exporting to image:
------
failed to solve: number of platforms does not match references 2 1
```

(Given how cryptic the BuildKit error is, it might also be beneficial [perhaps in `compose-go` loader/validate?] to error if duplicates exist in `service.build.platforms`, but I'm less concerned about that.)

Reply:
Also, is it weird that you can "extend" the supported platforms via `DOCKER_DEFAULT_PLATFORM`?

e.g. in the same example above, `DOCKER_DEFAULT_PLATFORM=linux/arm64 docker compose build hello` would build both AMD64 + ARM64 images even though only `linux/amd64` is actually in `compose.yaml`.

Reply:
Does it make sense for you if we manage the `DOCKER_DEFAULT_PLATFORM` like the `service.platform`, which means refuse to build if the `DOCKER_DEFAULT_PLATFORM` isn't defined in the `service.build.platforms` section?